File: server/models/Fine.js
const mongoose = require('mongoose');

const fineSchema = new mongoose.Schema({
    report_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'PollutionReport',
        required: true
    },
    issued_by: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    polluter_name: {
        type: String,
        default: null
    },
    polluter_contact: {
        type: String,
        default: null
    },
    fine_amount: {
        type: Number,
        required: true
    },
    fine_reason: {
        type: String,
        default: null
    },
    fine_category: {
        type: String,
        default: null
    },
    status: {
        type: String,
        enum: ['issued', 'paid', 'disputed', 'cancelled'],
        default: 'issued'
    }
}, {
    timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' },
    toJSON: { virtuals: true },
    toObject: { virtuals: true }
});

fineSchema.virtual('id').get(function () {
    return this._id.toHexString();
});

fineSchema.index({ report_id: 1 });

module.exports = mongoose.model('Fine', fineSchema);

File: server/models/fine.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class FineBase(BaseModel):
    report_id: str
    issued_by: str
    polluter_name: Optional[str] = None
    polluter_contact: Optional[str] = None
    fine_amount: float
    fine_reason: Optional[str] = None
    fine_category: Optional[str] = None
    status: str = "issued"

class FineResponse(FineBase):
    id: str
    created_at: datetime
    updated_at: datetime
